pages/api/polygon.js

```diff
@@ -0,0 +1,16 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+
+export default async function handler(req, res) {
+  const response = await fetch("https://api.polygonscan.com/api?module=account&action=txlist&address=" + req.query.wallet_address + "&startblock=0&endblock=99999999&page=1&offset=10&sort=asc&apikey=" + process.env.POLYGON_API_KEY);
+  const json = await response.json();
+
+  let hasUsedFaucet = false;
+
+  for (let i = 0; i < json.result.length; i++) {
+    if (json.result[i]["to"] == "0x67806adca0fd8825da9cddc69b9ba8837a64874b") {
```

Comment:
Do we want to be more explicit about what we're looking for here? Currently there's only one function to call so I think it's fine, but maybe worth double-checking.

Reply:
We need to make sure that the faucet call was successful and didn't error.

Example of a call that failed: https://polygonscan.com/tx/0x204e155350c3f06ebcce9c69c7edee0ab1ca5a99dcf30fe486bb9fe5327d1030

If it fails, do we catch that?

Reply:
Updated to check for failures.